src/pudl/analysis/allocate_gen_fuel.py

```diff
@@ -1659,19 +1660,11 @@ def adjust_msw_energy_source_codes(
     """
     # Adjust any energy source codes related to municipal solid waste
     # get a list of all of the MSW-related codes used in gf and bf
-    msw_codes_in_gf = set(
-        gf.loc[
-            gf["energy_source_code"].isin(["MSW", "MSB", "MSN"]),
-            "energy_source_code",
-        ].unique()
-    )
-    msw_codes_in_bf = set(
-        bf_by_gens.loc[
-            bf_by_gens["energy_source_code"].isin(["MSW", "MSB", "MSN"]),
-            "energy_source_code",
-        ].unique()
+    codes_used = set(gf.energy_source_code.unique()) | set(
+        bf_by_gens.energy_source_code.unique()
     )
-    msw_codes_used = list(msw_codes_in_gf | msw_codes_in_bf)
+    # NOTE 2025-12-11: we sort this reverse-ly to match the indeterminate order that happens to be in nightly builds right now
+    msw_codes_used = sorted({"MSN", "MSB", "MSW"} & codes_used, reverse=True)
```

Comment:
nit non-blocking preference for `intersection` instead of `&` bc it feels more explicit
```suggestion
    msw_codes_used = sorted({"MSN", "MSB", "MSW"}.intersection(codes_used), reverse=True)
```